src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -53,7 +53,20 @@ void AbstractTableGenerator::generate_and_store() {
    */
   {
     const auto& sort_order_by_table = _sort_order_by_table();
-    if (!sort_order_by_table.empty()) {
+    if (sort_order_by_table.empty()) {
+      // If there is no clustering for the benchmark defined, there should not be a single sorted chunk. This check is
```

Comment:
We should somewhere document the decision and the reasoning for not setting sorted-by-flags even for the columns that are sorted in the default output.

Reply:
Is that a real property? I thought is rather "some random fact" how data is generated. If it's not a stated property by TPC-H I think it could be rather confusing to say why we are not recognizing it.

Or did I get you wrong?

Reply:
Not sure if this was the point to leave the note. What I am saying is that we do not set table_clustered_by and in individual chunk order if we use the default order even though the data would allow us to.

Reply:
Added to the tpch table generator. Ok?
```
AbstractTableGenerator::SortOrderByTable TPCHTableGenerator::_sort_order_by_table() const {
  if (_clustering_configuration == ClusteringConfiguration::Pruning) {
    // This clustering improve the pruning of chunks for the two largest tables in TPC-H, lineitem and orders. Both
    // tables are frequently filtered by the sorted columns, which improves the pruning rate significantly.
    // Allowed as per TPC-H Specification, paragraph 1.5.2.
    return {{"lineitem", "l_shipdate"}, {"orders", "o_orderdate"}};
  }

  // Even though the generated TPC-H data is sorted by the primary keys, we do neither set the table to be clustered
  // or chunks to be sorted for any table. As of now, we do not automatically recognize these flags.
  return {};
}
```

Reply:
Mmh, "sorted" might be misleading in this context. Suggestion:

> Even though the generated TPC-H data is _implicitly_ sorted by the primary keys, we do neither set the corresponding flags in the table and in the chunks. This is done on purpose, as the non-clustered mode is designed to pass as little extra information into Hyrise as possible. In the future, these sort orders might be automatically identified with flags being set automatically.

Reply:
"nor in the chunks?

Reply:
yes